libgadget/utils/mymalloc.c

```diff
@@ -24,16 +25,36 @@ Allocator A_TEMP[1];
 #define allocator_init allocator_malloc_init
 #endif
 
+void
+tamalloc_init(void)
+{
+    int Nt = omp_get_max_threads();
+    int NTask;
+    MPI_Comm_size(MPI_COMM_WORLD, &NTask);
+
+    /* reserve 128 bytes per task and 128 bytes per thread for TEMP storage.*/
+    size_t n = 4096 * 1024 + 128 * NTask + 128 * Nt * NTask;
+
+    message(0, "Reserving %td bytes per rank for TEMP memory allocator. \n", n);
+
+    if (MPIU_Any(ALLOC_ENOMEMORY == allocator_init(A_TEMP, "TEMP", n, 1, NULL), MPI_COMM_WORLD)) {
+        endrun(0, "Insufficient memory for the TEMP allocator on at least one nodes."
+                  "Requestion %td bytes. Try reducing MaxMemSizePerNode. Also check the node health status.\n", n);
+
+    }
+}
+
 void
 mymalloc_init(double MaxMemSizePerNode)
```

Comment:
Is it worth it to change the interface to
mymalloc_init() {
ta_malloc_init() }

mymalloc_set_size(....) {
   ma_malloc_init(...)
}

and replace all mymalloc to ma_malloc().

Reply:
hmm, that seems a lot of effort. I don't think it is worth it.

Reply:
The point is after all to do ta_malloc before ma_malloc so the wrapper is not useful.